### PR TITLE
Feature/test enumerables

### PR DESCRIPTION
**Pull request by Mindo-Joseph**
#### What does this PR do?
Implements tests for enumerable methods in Ruby
#### Description of Task to be completed?
Write tests for the previously completed project on Ruby Enumerables
#### How should this be manually tested?
Clone the repository into your machine and inside the terminal run 'rspec'

### CODE REVIEW

File: spec/enumerables_spec.rb [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    # end
+  end
+  describe '#my_map' do
+    it 'if block is given returns a new array' do
+      expect([2, 3, 4].my_map { |i| i * 2 }).to eq([4, 6, 8])
+    end
+    it 'if block not given returns my_each ' do
+      allow([2, 3, 4]).to receive(:my_map).and_return(:my_each)
+    end
+  end
```
Comment (kobiyoyo) [inline spec/enumerables_spec.rb]: Let's write test for when a block is given

File: spec/enumerables_spec.rb [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    it 'returns true if no item matches a pattern set' do
+      expect([nil, true, 99].my_none?(String)).to eq true
+    end
+    it 'returns true when called on empty object' do
+      expect([].my_none?).to eq(true)
+    end
+    it 'returns true if all objects are falsey' do
+      expect([nil, false].none?).to eq(true)
+    end
+  end
```
Comment (kobiyoyo) [inline spec/enumerables_spec.rb]: Let's write test for when Regex is passed as an argument

File: spec/enumerables_spec.rb [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    it 'returns true if any item matches a pattern set' do
+      expect([nil, true, 99].my_any?(Integer)).to eq true
+    end
+    it 'returns false when called on empty object' do
+      expect([].my_any?).to eq(false)
+    end
+    it 'returns true if at least one object is truthy' do
+      expect([nil, true, 99].my_any?).to eq(true)
+    end
+  end
```
Comment (kobiyoyo) [inline spec/enumerables_spec.rb]: Let's write test for when Regex is passed as an argument

File: spec/enumerables_spec.rb [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    it 'returns false when it encounters any nil value' do
+      expect([nil, true, 99].my_all?).to eq(false)
+    end
+    it 'accepts a pattern' do
+      expect([1, 2i, 3.14].my_all?(Numeric)).to eq(true)
+    end
+    it 'returns true for an empty array' do
+      expect([].my_all?).to eq(true)
+    end
+  end
```
Comment (kobiyoyo) [inline spec/enumerables_spec.rb]: Let's write test for when Regex is passed as an argument

File: spec/enumerables_spec.rb [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  describe '#my_each_with_index' do
+    it 'belongs to the enumerator class' do
+      expect(arr.my_each_with_index.class).to eq(Enumerator)
+    end
+  end
+  describe '#my_select' do
+    it 'belongs to the enumerator class' do
+      expect(arr.my_select.class).to eq(Enumerator)
+    end
+  end
```
Comment (kobiyoyo) [inline spec/enumerables_spec.rb]: Let's write test for when a block is given

File: spec/enumerables_spec.rb [hunk 6]
```diff
@@ -0,0 +1,107 @@
+require './lib/enumerables.rb'
+
+RSpec.describe Enumerable do
+  let(:arr) { [1, 2, 3, 4, 5] }
+  describe '#my_each' do
+    it 'belongs to the enumerator class' do
+      expect(arr.my_each.class).to eq(Enumerator)
+    end
+  end
```
Comment (kobiyoyo) [inline spec/enumerables_spec.rb]: Let's write test for when a block is given

File: spec/enumerables_spec.rb [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+  describe '#my_each_with_index' do
+    it 'belongs to the enumerator class' do
+      expect(arr.my_each_with_index.class).to eq(Enumerator)
+    end
```
Comment (kobiyoyo) [inline spec/enumerables_spec.rb]: Let's write test for when a block is given

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team     
#### Great Job so far, but there are a few more things required. 
### General
- [ ]  Check comments below
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by maelfosso — APPROVED** [review]
# Congratulations :clap: 

Thank you for the changes done!
You can merge it now :+1: 
Good continuation :handshake: 

Mael FOSSO

**Pull request merged**